inst/@sym/function_handle.m

```diff
@@ -246,23 +253,27 @@
 %!test
 %! % output to disk
 %! fprintf('\n')
-%! f = function_handle(2*x*y, 2^x, 'vars', {x y z}, 'file', 'temp_test_output1');
+%! temp_file = tempname();
+%! temp_file(temp_file =='-') = '_';
```

Comment:
`temp_file = tempname('', 'oct_')`

I'm not sure its guaranteed to have to `-` but at least it doesn't by default ;-)